contracts/mocks/MockBinaryMerkleTree.sol

```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: Apache-2.0
-pragma solidity ^0.8.4;
+pragma solidity 0.8.9;
```

Comment:
I think we should keep this as `^0.8.4` (or whatever the minimum feature-version needed to compile and test) since this library is meant for external consumption by third parties as well, not just us. Restricting to a single Solidity version effectively completely prevents that.

Reply:
Agreed!